db/commitlog/commitlog.cc

```diff
@@ -741,7 +750,11 @@ class db::commitlog::segment : public enable_shared_from_this<segment>, public c
     friend class segment_manager;
 
     size_t buffer_position() const {
-        return _buffer.size_bytes() - _buffer_ostream.size();
+        // need some arithmatics to figure out what out actual positon is, including
+        // page checksums etc. The ostream does not include this.
+        auto size = _buffer_ostream.size();
+        size += (size / (_alignment - detail::sector_overhead_size)) * detail::sector_overhead_size;
```

Comment:
what about the remainder?
should we account for a partial sector at the tail?

Reply:
? That is what this does.

Reply:
I'd except `(size + (_alignment - detail::sector_overhead_size - 1) / (_alignment - detail::sector_overhead_size)) * detail::sector_overhead_size`

Reply:
The resulting size is aligned up to actual sector size on the line below. Here we just take into account the overhead caused by sector metadata.

Reply:
or something like:
```
auto d = std::div(size, _alignment - detail::sector_overhead_size);
size += (d.quot + bool(d.rem)) * detail::sector_overhead_size;
```

Reply:
> The resulting size is aligned up to actual sector size on the line below. Here we just take into account the overhead caused by sector metadata.

Ok, so the end result should be the same.
It's just odd that we count here the segment header overhead for only the full sectors but not for the last one, if it's partial.

At least, let's document that.

Reply:
We cover all overheads included in the size. Note the div is by actual data portion of a sector (not _alignment itself). The alignment after that is not new, we always align on disk write sector size.